<launch>
  <!-- <include file="$(find rtart_launch)/launch/includes/velocity_smoother.launch.xml"/>
  <include file="$(find rtart_launch)/launch/includes/safety_controller.launch.xml"/> -->
  
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="ranger_0/base_scan" />
  <arg name="tf_prefix" default="robot_0"/>
  <arg name="custom_param_file" default="$(find rl_dovs)/launch/move_base_params/dummy.yaml"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="tf_prefix"	value="$(arg tf_prefix)"/>
    <rosparam file="$(find rl_dovs)/launch/move_base_params/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find rl_dovs)/launch/move_base_params/costmap_common_params.yaml" command="load" ns="local_costmap" />   
    <rosparam file="$(find rl_dovs)/launch/move_base_params/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find rl_dovs)/launch/move_base_params/global_costmap_params.yaml" command="load" />
    <!-- <rosparam file="$(find rl_dovs)/launch/move_base_params/dwa_local_planner_params.yaml" command="load" /> -->
    <rosparam file="$(find rl_dovs)/launch/move_base_params/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find rl_dovs)/launch/move_base_params/move_base_params.yaml" command="load" />
    <rosparam file="$(find rl_dovs)/launch/move_base_params/global_planner_params.yaml" command="load" />
    <!-- <rosparam file="$(find rl_dovs)/launch/move_base_params/navfn_global_planner_params.yaml" command="load" /> -->
    <!-- external params file that could be loaded into the move_base namespace -->
    <!-- <rosparam file="$(arg custom_param_file)" command="load" /> -->
    
    <!-- reset frame_id parameters using user input data  -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg tf_prefix)/$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg tf_prefix)/$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg tf_prefix)/$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg tf_prefix)/$(arg odom_frame_id)"/>

    <remap from="map" to="/map" />
    <!-- <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/> -->
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="base_scan" to="$(arg laser_topic)"/>
  </node>
</launch>
